include/aspect/mesh_deformation/fastscapecc.h

```diff
@@ -0,0 +1,311 @@
+
+
+        /**
+         * Suggestion for the number of FastScape steps to run for every ASPECT timestep,
+         * where the FastScape timestep is determined by ASPECT_timestep_length divided by
+         * this parameter.
+         */
+        unsigned int fastscape_steps_per_aspect_step;
+
+        /**
+         * Maximum timestep allowed for FastScape, if the suggested timestep exceeds this
+         * limit it is repeatedly divided by 2 until the final timestep is smaller than this parameter.
+         */
+        double maximum_fastscape_timestep;
+
+
+        /**
+         * We want to check that every point within the FastScape mesh has been given a value.
+         * However, this is done only on processor 0, so this allows us to send an error later on
+         * if necessary.
+         */
+        // mutable bool fastscape_mesh_filled = true;
+
+        /**
+         * ASPECT end time to check if we should destroy FastScape.
+         */
+        double end_time;
+
+        /**
+         * FastScape cell size in X, dx should always equal dy.
+         */
+        double dx;
+
+        /**
+         * FastScape cell size in Y, dy should always equal dx.
+         */
+        double dy;
+
+        /**
+         * FastScape X extent (ASPECT X extent + 2*dx for ghost nodes).
+         */
+        double x_extent;
+
+        /**
+         * Fastscape Y extent (ASPECT Y extent + 2*dy for ghost nodes).
+         */
+        double y_extent;
+
+        /**
+         * User set FastScape Y extent for a 2D ASPECT model.
+         */
+        double y_extent_2d;
+
+        /**
+         * Number of x points in FastScape array.
+         */
+        int nx;
+
+        /**
+         * Number of y points in FastScape array.
+         */
+        int ny;
+
+        /**
+         * Size of the FastScape array (nx*ny).
+         */
+        unsigned int array_size;
+
+
+        /**
+         * How many levels FastScape should be refined above the maximum ASPECT surface resolution.
+         */
+        unsigned int additional_refinement;
+
+        /**
+         * Maximum expected refinement level at ASPECT's surface.
+         * This and resolution_difference are required to properly transfer node data from
+         * ASPECT to FastScape.
+         */
+        int maximum_surface_refinement_level;
+
+        /**
+         * Difference in refinement levels expected at the ASPECT surface,
+         * where this would be set to 2 if 3 refinement leves are set at the surface.
+         * This and surface_resolution are required to properly transfer node data from
+         * ASPECT to FastScape.
+         *
+         * TODO: Should this be kept this way, or make it so the input is the expected levels
+         * of refinement at the surface, and we can subtract one within the code? Also,
+         * it would be good to find a way to check these are correct, because they are a
+         * common source of errors.
```

Comment:
Yes, that is correct at the moment it is the biggest difference in refinement levels at the surface. I wasn't sure if it would be more intuitive to have the input parameter be this, or the number of refinement levels at the surface and we can then adjust this to still work with how the code is currently written.